src/com/sahokia/kos/concerttickets/model/Ticket.java

```diff
@@ -1,37 +1,42 @@
+package com.sahokia.kos.concerttickets.model;
+
+import com.sahokia.kos.concerttickets.interfaces.Printable;
+
 import java.math.BigDecimal;
 import java.security.SecureRandom;
 import java.time.LocalDateTime;
 import java.time.ZoneId;
 import java.text.DecimalFormat;
 import java.util.HashSet;
+import java.util.Objects;
 import java.util.Set;
 
 
-public class Ticket {
+public class Ticket extends Identifiable implements Printable {
     private static final Set<String> EXISTING_ID_S = new HashSet<>();
     private static final String CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
     private static final String DEFAULT_STRING_IF_EMPTY = "";
     private static final int ID_LENGTH = 4;
     private static final SecureRandom SECURE_RANDOM = new SecureRandom();
     private static final ZoneId ZONE_ID = ZoneId.of("Asia/Tbilisi");
-    private String id;
+
+    private final LocalDateTime creationDateTime;
     private String concertHall;
     private String eventCode;
-    private final long time;
+    private long time;
```

Comment:
why here long data type?)

Reply:
Changed the data type to int

Reply:
For time we have separate classes like Timestamp, LocalDateTime, etc

Reply:
Done